### ISSUE TITLE
Set set temperature

### ISSUE THREAD

**Issue by williamgeurts** [Hyundai-Kia-Connect/kia_uvo#800]
If I use the KIA app to start the airco I can set the temperature. If I use Homeassistant to start the airco the set temperature is fixed on 15 degrees, which was not enough to clean the front windows. 
Is a set temperature setting available in the API? And if so can you integrate it in HA? 
I am very happy with this integration, thanks for writing and maintaining the integration.

**Comment by cdnninja** [comment]
What temperature are you setting in home assistant when you call the service?

**Comment by williamgeurts** [comment]
I assume the sensor.set_temperature is the value. I don’t set it, it is a sensor and reports 15 degrees. Can I set a temperature when I call the service?Met vriendelijke groet,William GeurtsOp 11 jan 2024 om 22:10 heeft cdnninja ***@***.***> het volgende geschreven:﻿
What temperature are you setting in home assistant when you call the service?

**Comment by cdnninja** [comment]
You can set when you call the service.  

The sensor is current value.  If not running its lowest set point.

**Comment by williamgeurts** [comment]
Thanks,I installed it years ago but never used it until yesterday. Forgot the options.Met vriendelijke groet,William GeurtsOp 12 jan 2024 om 01:08 heeft cdnninja ***@***.***> het volgende geschreven:﻿
You can set when you call the service.
The sensor is current value.  If not running its lowest set point.

**Comment by williamgeurts** [comment]
Hi, I tested the service, some settings work, some settings don't work. Set Temperature and defrost works, however the back window and steering wheel is not (these options work in my KIA connect app). Also the seat heating is not working (but this is not an option in my KIA connect app). Als the duration is not working. I'll send my debug log. 
[kia.debug.log](https://github.com/Hyundai-Kia-Connect/kia_uvo/files/13931084/kia.debug.log)
This is my automation:
alias: Kia_Airco_aan
description: ""
trigger:
  - platform: state
    entity_id: input_boolean.kia_airco
condition: []
action:
  - service: kia_uvo.start_climate
    data:
      duration: 1
      heating: "4"
      climate: true
      device_id: XXXXXXXXXXXXXXXXXXX
      defrost: true
      temperature: 28
mode: single

**Comment by williamgeurts** [comment]
This issue is not closed, just clicked the wrong button. Sorry for that.

**Comment by cdnninja** [comment]
Only items that your car support will work. 

Any items working via native app that don't work in ha?

**Comment by williamgeurts** [comment]
Yes, the backwindow heating  and steering wheel works via the app. I also see that the airco automatically stops, but I cannot adjust that.Met vriendelijke groet,William GeurtsOp 14 jan 2024 om 13:10 heeft cdnninja ***@***.***> het volgende geschreven:﻿
Only items that your car support will work.
Any items working via native app that don't work in ha?

**Comment by cdnninja** [comment]
Work your way through heating values 1-6 to see which one gets the results you want.

**Comment by williamgeurts** [comment]
The heating value “1” seems to be the only possible value. With value 1 the front and back windows and the steering wheel are switched on. All other values result in all switched off of these functions. That works good enough for me.Seats are not heated, but that is also not available in the KIA app.Met vriendelijke groet,William GeurtsOp 14 jan 2024 om 20:43 heeft cdnninja ***@***.***> het volgende geschreven:﻿
Work your way through heating values 1-6 to see which one gets the results you want.

**Comment by cdnninja** [comment]
Does that match the mobile app? Or in the app can you toggle those separately.

**Comment by williamgeurts** [comment]
Yes, 2 options:1. Front window 2. Sight mirror, steeringwheel and back window.Due to a reconstruction of the road my car is parked somewhere else, it is difficult to test it. I will test it again within a few weeks.Met vriendelijke groet,William GeurtsOp 20 jan 2024 om 14:41 heeft cdnninja ***@***.***> het volgende geschreven:﻿
Does that match the mobile app? Or in the app can you toggle those separately.
